src/components/AttachmentPicker/index.native.js

```diff
@@ -203,6 +192,25 @@ class AttachmentPicker extends Component {
         this.setState({isVisible: false});
     }
 
+    /**
+     * Setup native attachment selection to start after this popover closes
+     *
+     * @param {{pickAttachment: function}} item - an item from this.menuItemData
+     */
+    selectItem(item) {
+        /* setTimeout delays execution to the frame after the modal closes
+        * without this on iOS closing the modal closes the gallery/camera as well */
+        this.onModalHide = () => setTimeout(
+            () => item.pickAttachment()
+                .then(this.pickAttachment)
+                .catch(console.error)
+                .finally(() => delete this.onModalHide),
+            10,
+        );
+
+        this.close();
+    }
```

Comment:
Unfortunately on iOS the closing of the modal dismisses the gallery/documents picker 
So if they are triggered to appear even a split second before this popup closes they are automatically dismissed

---

Both the image picker and the document picker are behind their major version releases. There's already a ticker regarding updating the image picker library to v3 as v2 is no longer supported. I think that on the updated libraries the `setTimeout` would not be needed

Reply:
Could you please create a follow-up GH issue to remove the `setTimeout` once the libraries are updated and this is no longer a problem?

Reply:
Tagging @tgolen due to the `setState` call 

I couldn't find a way around this. 
If you launch the camera/gallery and then close the modal it instantly closes the gallery as well, but even when it's set up with `onModalHide` it would still sometimes close the gallery

I know we've dealt with this before here #2114 I did a lot of research back then and found out the problem does not happen if you trigger another modal, so we implemented the AttachmentPicker as a modal like the CreateMenu and replaced the ActionSheet used from `react-native-image-picker`

Now the same issue happens if we try to launch the gallery too soon again iOS only

Reply:
The only way to not use `setTimeout` is to keep the second modal open the entire time - which is the exact opposite of what this PR tries to fix

Reply:
> Could you please create a follow-up GH issue to remove the `setTimeout` once the libraries are updated and this is no longer a problem?

I would, though I'm not 100% sure this will allow the removing the `setTimeout` call 
What happens is when a modal opens it attaches to some native modal view, when the gallery is launched it seems it attaches to the same view and the closing of the modal destroys that view interpreted as user cancel
I've tried updating the modal library to the latest version - it didn't help...

Anyway I would at least make a ticket about it

Reply:
OK, thanks for the info! We can always revisit it later after the libs are updated anyway.

Reply:
Created a follow-up ticket: #2719